src/components/BounceRuleDetailed/Modals/index.jsx

```diff
@@ -0,0 +1,188 @@
+      <Column width={1} offset={10}>
+        <Button
+          className="sg-button"
+          onClick={handleModalClose}
+          id={modalType}
+          type="secondary"
+        >
+          {"Close"}
+        </Button>
+      </Column>
+      <Column width={1} offset={11}>
+        <Button
+          className="sg-button"
+          id={modalType}
+          onClick={handleModalConfirm}
+          type="primary"
+        >
+          {"Confirm"}
+        </Button>
+      </Column>
+    </Row>
+  </div>
+);
+
+const ChangeModalBody = ({ currentRule, handleModalClose }) => (
+  <div className="changelog-modal">
+    <Row>
+      <Column width={6} offset={1}>
+        <h1>Current</h1>
+        <ChangeTable currentRule={currentRule} />
+      </Column>
+      <Column width={6} offset={7}>
+        <h1>Previous</h1>
+        <ChangeTable currentRule={currentRule} />
+      </Column>
+    </Row>
+    <Row>
+      <Column className="changelog-modal-button sg-right" width={1} offset={12}>
+        <Button id="changeModal" onClick={handleModalClose}>
+          Close
+        </Button>
+      </Column>
+    </Row>
+  </div>
+);
+
+const ChangeTable = ({ currentRule }) => {
+  const {
+    description,
+    id,
+    response_code: responseCode,
+    enhanced_code: enhancedCode,
+    bounce_action: bounceAction,
+    regex,
+    priority,
+  } = currentRule;
+  return (
+    <Table className="change-table">
+      <TableBody>
+        <TableRow>
+          <TableCell>
+            <strong>ID</strong>
+          </TableCell>
+          <TableCell>{id}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Bounce Action</strong>
+          </TableCell>
+          <TableCell>{bounceAction}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Response Code</strong>
+          </TableCell>
+          <TableCell>{responseCode}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Enhanced Code</strong>
+          </TableCell>
+          <TableCell>{enhancedCode}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>RegEx</strong>
+          </TableCell>
+          <TableCell>{regex}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Priority</strong>
+          </TableCell>
+          <TableCell>{priority}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Description</strong>
+          </TableCell>
+          <TableCell>{description}</TableCell>
+        </TableRow>
+      </TableBody>
+    </Table>
+  );
+};
+
+const ModalDisplay = ({
+  isChangeModalOpen,
+  isConfirmOpen,
+  isCancelConfirmOpen,
+  currentRule,
+  handleModalClose,
+  handleModalConfirm,
+}) => (
+  <CenterModal
+    large={isChangeModalOpen}
+    open={isChangeModalOpen || isConfirmOpen || isCancelConfirmOpen}
```

Comment:
can wrap all these bools here in one variable like isCenterModalOpen = isChangeModalOpen || isConfirmOpen || isCancelConfirmOpen. Now that I think about it, since there is a lot of ||s for deciding how it will be open or which body should be showing. it may be nicer to break out each of these modals into their own components to avoid all the nested ors and conditions and we can break out stories even nicer by files and variations of the specific modal we're looking for. It'll help to maintain this in the long run with separate modals rather than all the conditions inside the renderBody. rather than different modal body components why not different modals in general that encapsulate the bodies already?